### PR TITLE
Add JetStream documentation

### PR DESCRIPTION
**Pull request by variadico**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This change adds NATS JetStream documentation. In addition, it adds a [deprecation warning](https://github.com/nats-io/nats-streaming-server/#warning--deprecation-notice-warning) to the NATS Streaming PubSub component.

## Issue reference

This PR is an accompaniment for PR https://github.com/dapr/components-contrib/pull/1101.

### CODE REVIEW

**Review by greenie-msft — APPROVED** [review]

File: daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ Table captions:
 | [Hazelcast]({{< ref setup-hazelcast.md >}})           | Alpha  | v1 | 1.0 |
 | [MQTT]({{< ref setup-mqtt.md >}})                     | Alpha  | v1 | 1.0 |
 | [NATS Streaming]({{< ref setup-nats-streaming.md >}}) | Beta  | v1 | 1.0 |
+| [JetStream]({{< ref setup-jetstream.md >}}) | Beta  | v1 | 1.4 |
```
Comment (yaron2) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md]: All components start as Alpha, please change.

**Review by yaron2 — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md (hunk 1 above)
Comment (variadico) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md]: Ah, whoops. I've changed this to Alpha. https://github.com/dapr/docs/pull/1751/commits/0d46589f18593d4b4395ccfcd88f27a569dc2ece

**Review by yaron2 — APPROVED** [review]

**Comment by orizohar** [comment]
@variadico - Looks like this is a new component that will be added in v1.4. Please change this PR to merge in to v1.4 branch of the docs

**Comment by variadico** [comment]
Sure thing, base updated. 👍🏽

**Review by greenie-msft — APPROVED** [review]

**Review by orizohar — APPROVED** [review]

**Pull request merged**